### PR TITLE
[feat] Cleanup repo orphan params/sequences

### PR DESCRIPTION
**Pull request by mihran113**

### CODE REVIEW

File: aim/cli/storage/commands.py [hunk 1]
```diff
@@ -113,3 +114,69 @@ def restore_runs(ctx, hashes):
...
+
+    # everything left in the `repo_paths` set is subject to be deleted
+    if not repo_paths:
+        click.echo('No orphan params were found')
+        return
+
+    # acquire index container to delete orphan paths
+    index_tree = repo._get_index_tree('meta', timeout=5).subtree('meta')
+
+    # start deleting with the deepest paths first to bypass the cases when parent path is deleted before the child
```
Comment (alberttorosyan) [inline aim/cli/storage/commands.py:180]: would it be too complex to optimize this part? i.e. if there are paths `('a', 'b', 'c'),  ('a', 'b', 'd'), ('a', 'b', 'e'), ...` and path `('a', 'b')`, delete just `('a', 'b')`

File: docs/source/refs/cli.md [hunk 2]
```diff
@@ -200,6 +200,7 @@ __storage subcommands__
 | `upgrade 2to3`   | Upgrades legacy Aim repository from `2.x` to `3.0`.                                      |
 | `upgrade 3.11+`  | Update metric sequence data format for given runs. At least one run should be specified. |
 | `restore`        | Rollback `Run` to old metric format if run backup is available.                          |
+| `prune`          | Remove orphan params/sequences with no referring runs                                    |
```
Comment (alberttorosyan) [inline docs/source/refs/cli.md:203]: worth describing what we mean by saying "orphan"

Comment (alberttorosyan) [reply docs/source/refs/cli.md:203]: maybe even a small section in docs, describing a use-model (i.e. run moved/deleted, params overwritten, etc.)

**Review by alberttorosyan — COMMENTED** [review]
overall looks good!
left a couple of comments

also lets test on a relatively big repo to see how long does the command take to run

File: aim/cli/storage/commands.py (hunk 1 above)
Comment (mihran113) [reply aim/cli/storage/commands.py:180]: Can't think of a relatively simple approach now, also I don't think that it's worth the trouble, as we're doing this operation on a single `rocksdb` container, shouldn't take too long I guess

File: docs/source/refs/cli.md (hunk 2 above)
Comment (mihran113) [reply docs/source/refs/cli.md]: done ✅

**Review by alberttorosyan — APPROVED** [review]
lgtm

**Comment by mihran113** [comment]
Resolves #1684

**Pull request merged**